libs/framework-core/source/ftrack_framework_core/host/__init__.py

```diff
@@ -265,7 +266,7 @@ def run_tool_config_callback(self, event):
             )
             engine_instance = engine_registry['extension'](
                 self.registry,
-                self.session,
+                session or self.session,
```

Comment:
I think we'd like to ensure that we're always starting a new session. In that case we should not fall back to an existing sessions if the creation of a new session fails. We should probably assert whether the new session is valid and raise an exception otherwise (or at least log an error to not crash the application).